### ISSUE TITLE
Fails to parse JSON, doesn't give us the raw response.

### ISSUE THREAD

**Issue by ponelat** [forwardemail/superagent#675]
For particular reasons, the response from the server is a malformed JSON string, ie: it will fail to parse.
However we'd still like to show the raw body content but can't because superagent doesn't include that data.

Can we have the err callback included the raw response.

Round about here I think...
https://github.com/visionmedia/superagent/blob/master/lib/client.js#L466-L473

**Comment by black-perl** [comment]
Yeah, I am having the same problem. Actually superagent checks the `Content-Type` of the returned `response`. If the content is `text` itself but if the content type is wrong like `application/json` , it tries to parse it. If an error occurs during parsing, `double callback error` is thrown and `res` is not set.

**Comment by catc** [comment]
Same issue here. Response from API is plain text but the header is `text/json` which causes superagent to throw the `double callback` error.

**Comment by black-perl** [comment]
There should be some option to get the raw response text in case the JSON parsing fails ?

**Comment by reinaldo13** [comment]
I added a `.rawResponse` property to the `err` object if the parsing fails. The code is pending approval in pull request https://github.com/visionmedia/superagent/pull/716

**Comment by black-perl** [comment]
@reinaldo13  Great !

**Comment by FoghostCn** [comment]
I had the same problem, content-type is application/json but failed to parse,then it warned double callback!

**Comment by antwan1986** [comment]
Same here; I'm purposefully sending back malformed JSON (as that's what numerous Google scripts do to stop people injecting the resource directly onto a apage in a `script` tag). 

A way to either disable the auto-parsing, or, somehow get the response in the error would allow me to skirt around this problem.

**Comment by pornel** [comment]
#716 added `rawResponse` to the error. Does that solve your problem?

**Comment by antwan1986** [comment]
It'll certainly give a workaround for examining the error and seeing if it "really" is an error. Having some way to disable the auto-parsing sounds like it'd be pretty useful but perhaps the use cases for this functionality are too small.

Do you know when this PR will be included into a release or SuperAgent that could be pulled over NPM? When's your next release?

**Comment by pornel** [comment]
I'll be released soon

**Comment by titanism** [comment]
Malformed JSON is supported: the original body is exposed as error.rawResponse, including browser paths, and regression tests cover it. This was addressed by #716.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.
